Classes/SoundFileAnalysis.sc

```diff
@@ -49,18 +49,33 @@ SoundFileAnalysis {
 		^result
 	}
 
-	analyzeFile { |path, start = (0), duration, which, callback, maxDataPoints = (1000)|
+	analyzeFile { |path, start = (0), duration, which, callback, maxDataPoints = (1000),trimDuration=\roundDown, serverOptions|
 		var result = ();
 		if(File.exists(path).not) { "\nFile not found: %\n".format(path).warn; ^this };
 		which = (which ?? { analysisMethods.keys.as(Array).sort }).asArray;
 		fork {
 			var resultpaths, oscpath, score;
-			var analysisDuration, soundFile, cond;
+			var analysisDuration, soundFile, options, cond;
 			var server = Server(\dummy);
 
 			// get duration and numChannels from soundFile
 			soundFile = SoundFile.openRead(path);
-			analysisDuration = min(duration ?? { soundFile.duration - start }, soundFile.duration);
+
+			options= (serverOptions?ServerOptions()
```

Comment:
could you keep whitespaces between operators everywhere, please? (sorry!) Also, it saves some cpu cycles if you write: `(serverOptions ?? { ServerOptions() ` here.